File: project-euler/problem 3/solution.py
def prime_factor(number):
    largest_factor = 1
    c = number
    i = 2

    while c > 1:
        if c % i == 0:
            largest_factor = i
            c = c // i
        i += 1
    print(largest_factor)


prime_factor(600851475143)

File: project-euler/problem 3/solution.dart
void main(List<String> args) {
  int largestPrimeFactor = primeFactor(600851475143);

  print(largestPrimeFactor);
}

int primeFactor(int number) {
  int largestFactor = 1;
  int c = number;
  int i = 2;

  while (c > 1) {
    if (c % i == 0) {
      largestFactor = i;
      c = (c / i).floor();
    }
    i += 1;
  }

  return largestFactor;
}
